bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceDialog.java

```diff
@@ -1336,15 +1345,18 @@ private void activateInFindReplaceLogicIf(SearchOptions option, boolean shouldAc
 	 * Evaluate the status of the FindReplaceLogic object.
 	 */
 	private void evaluateFindReplaceStatus() {
-		IFindReplaceStatus status = findReplaceLogic.getStatus();
+		decorate();
+	}
 
-		String dialogMessage = status.accept(new FindReplaceLogicMessageGenerator());
-		fStatusLabel.setText(dialogMessage);
-		if (status.isInputValid()) {
-			fStatusLabel.setForeground(fReplaceLabel.getForeground());
-		} else {
-			fStatusLabel.setForeground(JFaceColors.getErrorText(fStatusLabel.getDisplay()));
-		}
+	public void decorate() {
```

Comment:
If this one is only called once, please inline it into the caller _i.e._ remove the new method and put its body into the only caller.

Also, why `public` and not `private`?

Reply:
done